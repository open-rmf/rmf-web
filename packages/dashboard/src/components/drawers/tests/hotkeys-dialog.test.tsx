import { shallow } from 'enzyme';
import React from 'react';
import { GlobalHotKeys, KeySequence } from 'react-hotkeys';
import HotKeysDialog from '../hotkeys-dialog';

describe('Hotkeys Dialog', () => {
  test('Component renders correctly', () => {
    const keyMap = {
      OPEN_COMMANDS: {
        name: 'Open Commands',
        sequences: [{ sequence: 'shift+c', action: 'keypress' }],
      } as KeySequence,
    };

    const handlers = {
      OPEN_COMMANDS: () => jest.fn(),
    };

    const root = shallow(
      <GlobalHotKeys keyMap={keyMap} handlers={handlers}>
        <HotKeysDialog handleClose={() => jest.fn()} open={true} />
      </GlobalHotKeys>,
    );
    const hotKeyDialog = root.find(HotKeysDialog).dive();
    expect(hotKeyDialog.find('[detail-type="hotkeyDetail"]').length).toBe(
      Object.keys(keyMap).length,
    );
    expect(root.find(HotKeysDialog).dive()).toMatchSnapshot();
  });
});
