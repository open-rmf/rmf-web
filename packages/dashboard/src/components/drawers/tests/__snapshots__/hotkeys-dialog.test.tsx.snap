// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotkeys Dialog Component renders correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="customized-dialog-title"
  fullWidth={true}
  maxWidth="sm"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="customized-dialog-title"
  >
    <div>
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        Hotkeys
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="close"
        className="makeStyles-closeButton-1"
        onClick={[Function]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))
    className="makeStyles-dialogContent-2"
    dividers={true}
  >
    <div
      className="makeStyles-detail-5"
    >
      <div
        className="makeStyles-detailLine-4"
        detail-type="hotkeyDetail"
      >
        <WithStyles(ForwardRef(Typography))
          variant="body1"
        >
          Open Commands
          :
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="body1"
        >
          shift+c
        </WithStyles(ForwardRef(Typography))>
      </div>
      <WithStyles(ForwardRef(Divider)) />
    </div>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))
    className="makeStyles-dialogActions-3"
  >
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      color="primary"
      onClick={[Function]}
    >
      OK
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
